Getting started with windows forms using c#

mohamedgh16 opened the pull request:
### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

mohamedgh16 commented:
this fixes #913

hectorkambow commented:
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

hectorkambow commented:
plagiarism check done

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,169 @@
+```
+
+In this piece of code, we will deal with the `Sqrt()` & the `Pow()` methods.
+**NOTE:** that In order To use the Pow & Sqrt methods you need to import the Math class as follows `using static System.Math;`.
````
If you using `NOTE:`, then you should not start the sentence after the `:` like it's a continuation of the Note.

You can rewrite as,

NOTE: In order to use ...

zolomohan replied:
Fix at all places in the article.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,169 @@
+        }
+```
+
+Finally, we will work on the Calculate button to evaluate the string expression, and Don't forget to check for invalid operations or inputs.
````
Remove capitalization of Don't and fix all instances like this in the article.

zolomohan commented:
@mohamedgh16,
You missed explaining some steps in the tutorial. You must explain all the steps required, step by step for everyone to understand.

I've marked all the places where further explanation is needed.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,169 @@
+### Building the form with the controls
+First of all, let's start with the form and change the text to My calculator. **NOTE:** that changing the name of a Control or the form will only change the name of the object inside the code, and that makes it easier to figure out which button or witch textbox you are using or editing right now, on the other hand changing the text of a Control will only change what appears to the user.
+
+Now let's begin by adding the buttons and the textbox. First, we will add the numbers as buttons then we will add the C button the Calculating button the Operation buttons, and finally, the Textbox that will show the answer.
+
+**NOTE:** don't forget to change both the controls name and the text.
```
You must explain for Textbox separately.

mohamedgh16 commented:
@zolomohan Thanks for the notes! I have done the changes.
Let me know if there is anything else to do.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+**Windows forms** allow the programmers to create a graphical user interface (GUI) that allows the interaction with the program visually. According to [Microsoft](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/overview/?view=netdesktop-5.0), It provides one of the most productive ways to create desktop apps based on the visual designer provided in Visual Studio. Functionality such as dr ...
```
Rewrite as.

Functionality like the drag-and-drop placement of controls makes it easy to build the GUI.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+
+With *windows forms*, you can make basic applications like a calculator to high-level applications such as a library database management application that requires a connection to the database from the application.
+### Prerequisites
+1. A basic understanding of c# programming language.
```
Capitalize C in C#

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+1. A basic understanding of c# programming language.
+2. Visual studio installed on your system.
+
+If you don't have Visual Studio installed on your computer. you can download it from [Here](https://visualstudio.microsoft.com/downloads/).
```
Instead of just providing the link for the Visual Studio code, can you provide an article or documentation page that explains how to set up the C# environment in Visual Studio.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+If you don't have Visual Studio installed on your computer. you can download it from [Here](https://visualstudio.microsoft.com/downloads/).
+
+### Make a new project
+In this tutorial, we will make a simple calculator! Make a new project by opening the visual studio, then click on File -> New -> Project -> Visual c# -> Windows desktop and choose Windows Forms App (.NET Framework).
```
Replace `!` with `.`

zolomohan replied:
Capitalize Visual Studio.

zolomohan replied:
Capitalize C in C#

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+The *Toolbox* window provides you with all the controls that help you build your project inside the form. to display the Toolbox click on View -> Toolbox, or you can press Ctrl+Alt+X. To use one of the controls double click on one of them or simply drag and drop it.
+
+### Properties window
+The *Properties window* is used to show the properties of the form or the control that you click on, you can manipulate the properties of any control used in your form. To display the window you can click on View -> properties window or you can simply click F4.   
```
The *Properties window* is used to show the properties of the form or the control that you click on,

Instead of a comma, break the sentence with a period (`.`)

zolomohan replied:
Instead of `To display the window...`, be specific and say `To open the properties window, ...`

Don't forget to add a comma.

zolomohan replied:
Make sure you capitalize on all instances after `->`

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+![properties window and toolbox](properties window and toolbox.png)
+
+### Building the form with the controls
+First of all, let's start with the form and change the text to My calculator and the name to Cal_Form from the properties window.
+
+![Form_name_text](Form_name_text.png).
+
+
+**NOTE** that changing the name of a Control or the form will only change the name of the object inside the code, and that makes it easier to figure out which button or witch textbox you are using or editing right now, on the other hand changing the text of a Control will only change what appears to the user.
```
Move the NOTE above the image.

zolomohan replied:
Break sentences with a period rather than a comma.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+
+**NOTE** that changing the name of a Control or the form will only change the name of the object inside the code, and that makes it easier to figure out which button or witch textbox you are using or editing right now, on the other hand changing the text of a Control will only change what appears to the user.
+
+Now let's begin by adding the buttons and the textbox. First, we will add the numbers as buttons then we will add the C button the Calculating button the Operation buttons, and finally, the Textbox that will show the answer. To add a Control, you can drag & drop it from the toolbox to the form or simply double click on it. Repeat this with remaining controls.
```
Add Commas between items and fix the punctuation. There's a lot of unnecessary capitalization.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+
+![Controls_display](Controls_display.png).
+
+Don't forget to change both the controls *Texts* to numbers & operations and the *Name* of buttons & textbox that will change the object name inside the code like we previously did with the form.
```
Instead of saying Don't forget, you can instruct the reader to change the texts and name of the controls.

zolomohan replied:
The sentence is too long. Please break it down.

zolomohan replied:
You should also specify what **name** you are using for the buttons and the text box.

You don't want the reader to give the controls their own name and find out you are using something different.
Worst case scenario, they might get confused with your names that you have specified in the code.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+![First_look1](First_look1.png).
+
+### Let's start coding
+You can double click on any of the buttons and it will take you to the method where you can write your code inside of it, let's start with the numbers.
```
Let's start with numbers should be a separate line. It should now come after a comma.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+
+### Let's start coding
+You can double click on any of the buttons and it will take you to the method where you can write your code inside of it, let's start with the numbers.
+Each one of these buttons will display a number on the textbox that we previously added.
```
You can explain it better. Like,

When you click a button, the value of the button must get appended to the text box's value, which we will use later for calculating. 

Once we update the value of the textbox, it will be visually updated in the textbox.

zolomohan replied:
Something like this.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+You can double click on any of the buttons and it will take you to the method where you can write your code inside of it, let's start with the numbers.
+Each one of these buttons will display a number on the textbox that we previously added.
+
+**NOTE** that the textbox's *Name* is changed to `Output.Text`.
```
Move this line to the previous section.

You can specify here that you need to use the Textbox's name to update it's value.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+
+Repeat this piece of code with all the buttons that produce numbers.
+
+Inside the `Dot_Click` we should handle the case of multiple dots. To fix this problem, we should check the `Output.Text` if it has a dot already, we will prevent the user from adding another dot, if not, we can normally add a dot to the text. **NOTE** that we use the `try()` & `catch()` In order To prevent the user from doing invalid operations.
```
Break the sentence with a period after `we should check the Output.Text`

zolomohan replied:
You specified try and catch here, but you not using it in the below code snippet. Remove it if it's unnecessary or if it's misplaced.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,176 @@
+}
+```
+
+Repeat this piece of code with all the buttons that produce numbers.
+
+Inside the `Dot_Click` we should handle the case of multiple dots. To fix this problem, we should check the `Output.Text` if it has a dot already, we will prevent the user from adding another dot, if not, we can normally add a dot to the text. **NOTE** that we use the `try()` & `catch()` In order To prevent the user from doing invalid operations.
````
You only instructed the readers to add the buttons for numbers. You didn't instruct them to add a button for `.`

Add a line stating so.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,176 @@
+}
+```
+
+To clear the `Output.Text` we simply do this.
````
Instruct the reader to add a button to clear the textbox.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,176 @@
+}
+```
+
+Now let's move to the operations.
````
Instruct the reader to add a button for all the operations.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,176 @@
+}
+```
+
+In this piece of code, we will deal with the `Sqrt()` & the `Pow()` methods.
+**NOTE**: To use the Pow & Sqrt methods you need to import the Math class as follows `using static System.Math;`.
+We will also use the `try()` & `catch()` to prevent the user from doing invalid operations.
````
Instruct the reader to add a button for power and square root.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
````diff
@@ -0,0 +1,176 @@
+```
+
+### Final touches
+Let's change some colors and fonts using the properties window!
````
Replace `!` with `.`

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,176 @@
+![Final_look](Final_look.png).
+
+### Conclusion
+In this tutorial, we created a basic calculator using windows forms, we have learned how easy it is to build various applications with it using very flexible Controls,then we learned how to connect these Controls to the code inside of it. However, this is not over! We will build more applications using windows forms in the upcoming tutorials. Don’t forget to test out the code to fully understand  ...
```
Break this down into points.

zolomohan reviewed (changes requested):
@mohamedgh16, I've left some suggestions for you. Ping me when you have implemented them.

mohamedgh16 commented:
@zolomohan I have done the changes.

about the buttons, they are already added, but to make things more clear for the reader I have added an explanation sentence. 
(Note that you have to double click on each control to start coding inside its method. Do this with all the following controls).

I have also added a table that contains the name of all the controls used.

let me know if there's anything else.

zolomohan commented on articles/Getting-started-with-windows-forms-using-c-#/index.md:
```diff
@@ -0,0 +1,197 @@
+### Properties window
+The *Properties* window is used to edit the properties of the form or the control (button, textbox, etc.) that you click on. To open the properties window, you can click on View -> Properties window or you can simply click F4.   
+
+![properties window and toolbox](properties window and toolbox.png)
```
This image is not loading because the filename has spaces. Can you please rename the file and update it here?

zolomohan replied:
We can move to the final review if this is fixed.

mohamedgh16 replied:
@zolomohan Done.

hectorkambow commented:
Great article @mohamedgh16  

& great revision at @zolomohan  

@mohamedgh16  Be sure to implement any (applicable) changes to future articles to continue enhancing the quality of the articles. 🚀
